src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,426 @@
+            {ManagedKafkaType.standard},
+            {ManagedKafkaType.developer}};
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReservedDeploymentExistence(ManagedKafkaType mkType) {
+
+        String namespace = String.format("reserved-kafka-%s-1", mkType);
+        log.info("evaluating namespace '{}'", namespace);
+
+        String managedKafkaName = String.format("reserved-kafka-%s-1", mkType);
+        log.info("managed kafka reserved deployment '{}'", managedKafkaName);
+        Optional<ManagedKafka> mkOptional = FleetshardUtils.managedKafka(oc).inNamespace(namespace).list().getItems().stream().filter(ManagedKafka::isReserveDeployment).findAny();
+        Assert.assertTrue(mkOptional.isPresent(), "reserved deployment is not present");
+
+        log.info("assert existence of resources which are part of reserved deployment");
+        int expectedZookeeperPodsCount;
+        int expectedKafkaPodsCount;
+        switch (mkType) {
+            case standard:
+                expectedZookeeperPodsCount = 3;
+                expectedKafkaPodsCount = 3;
+                break;
+            case developer:
+                expectedZookeeperPodsCount = 1;
+                expectedKafkaPodsCount = 1;
+                break;
+            default:
+                throw new Exception("unsupported managed kafka type");
+        }
+
+        log.info("test that '{}' Zookeeper Pod/s are present in namespace '{}'", expectedZookeeperPodsCount, namespace);
+        int actualNumberOfZookeeperPods = countNumberOfPodsWithNameExisting(namespace, "zookeeper");
+        Assert.assertEquals(expectedZookeeperPodsCount, actualNumberOfZookeeperPods, "unexpected number of zookeeper pods");
+
+        log.info("test that '{}' Kafka Pod/s are present in namespace '{}'", expectedKafkaPodsCount, namespace);
+        int actualNumberOfKafkaPods = countNumberOfPodsWithNameExisting(namespace, "kafka");
+        Assert.assertEquals(expectedKafkaPodsCount, actualNumberOfKafkaPods, "unexpected number of kafka pods");
+
+    }
+
+    private int countNumberOfPodsWithNameExisting(String namespace, String nameSubstring) {
+        log.debug("get pods count with nameSubstring '{}'  in namespace '{}'", nameSubstring, namespace);
+        var podsPresent =  oc.pods().inNamespace(namespace).list().getItems().stream()
+            .filter(e -> e.getMetadata().getName().contains(nameSubstring))
+            .collect(Collectors.toList());
+        log.debug(podsPresent);
+        return podsPresent.size();
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    public void testReservedDeploymentDoesNotPreventKafkaCreation(ManagedKafkaType mkType) throws Exception {
+
+        log.info("testing managed kafka type: '{}'", mkType);
+
+        // obtain max limit of kafka instances per given mk type (developer, standard)
+        int upperStreamingUnitLimitPerManagedKafkaType;
+        switch (mkType.name()) {
```

Comment:
no need to convert to a string. You can have the switch statement on the enum.

Reply:
yes in this case it makes complete  sense to go with ManagedKafkaType rather than string param.